### ISSUE TITLE
Unable to build the module on MacOS-Catelina

### ISSUE THREAD

**Issue by binpetro** [OPM/opm-upscaling#320]
Hi, 
I am trying to build OPM for MacOS. I'm unable to compile the Upscaling route. I tried follow the method presented in the OPM installing from source page on the website. 
The error as
```
too many template arguments for class template 'LeafMultipleCodimMultipleGeomTypeMapper'
```
This pops up while processing the file `opm/upscaling/RelPermUtils.cpp`. Just for reference I am including CmakeCache, Error log files. 

[CMakeCache.txt](https://github.com/OPM/opm-upscaling/files/4750707/CMakeCache.txt)
[CMakeError.log](https://github.com/OPM/opm-upscaling/files/4750708/CMakeError.log)
[CMakeOutput.log](https://github.com/OPM/opm-upscaling/files/4750712/CMakeOutput.log)

Any suggestion and help would be appreciated.
Thanks everyone in advance for your suggestion.

**Comment by binpetro** [comment]
Any suggestion on how to compile this module in Mac?

**Comment by atgeirr** [comment]
Sorry for the late answer. I am not enough of a CMake guru to read your log files and figure out what happened, but if you could give me the complete error message, with file and line number, I could perhaps find out something. It compiles fine for me (on a mac). Also, what is the Dune version you are using?

**Comment by bska** [comment]
Are you by any chance using the Git master/unstable sources of Dune?

Apparently one of the template arguments in [`Dune::MultipleCodimMultipleGeomTypeMapper`](https://gitlab.dune-project.org/core/dune-grid/-/blob/master/dune/grid/common/mcmgmapper.hh#L126-128) was removed after the 2.7 Dune release.  The opm-upscaling module targets specific releases of Dune&mdash;currently 2.6 with 2.7 support being spotty.

We do not currently attempt to track the development sources of Dune and that does mean we occasionally get build breaks when someone tries to target that version with OPM modules.

> Any suggestion on how to compile this module in Mac?

At this point, if you do use the Dune master development sources, I don't have a better suggestion than to fall back to a fixed release, typically 2.6 or maybe 2.7 if you're adventurous.  If you are **not** using the Dune master sources from git then we need to look into this more deeply.

**Comment by binpetro** [comment]
> Sorry for the late answer. I am not enough of a CMake guru to read your log files and figure out what happened, but if you could give me the complete error message, with file and line number, I could perhaps find out something. It compiles fine for me (on a mac). Also, what is the Dune version you are using?

I am using the Master version of Dune and OPM. The error is 
/opm/porsol/common/GridInterfaceEuler.hpp:71:23: error: too many
      template arguments for class template 'LeafMultipleCodimMultipleGeomTypeMapper'

If the dune -version is an issue, I might have to run the whole build back again with 2.6 version. Please suggest what could be a best solution.

**Comment by atgeirr** [comment]
> If the dune -version is an issue, I might have to run the whole build back again with 2.6 version. Please suggest what could be a best solution.

I think using the 2.6 release is the simplest solution. On the other hand, if you are able to update OPM to work with the latest Dune that would be appreciated, and a valuable contribution!

**Comment by blattms** [comment]
I think this is fixed by my recent MR (DUNE 2.8). Closing. If the problem persists then don't hesitate to reopen.
